### PR TITLE
Fix for Missing Class features

### PR DESCRIPTION
**Pull request by dmportella**
Added fix for https://github.com/kakaroto/Beyond20/issues/1164

Tested in chrome, and mobile, tablet and desktop layouts

After changes code now correctly shows the Genie Wrath and class features are populated (Needs verifying ALL features are working, will look into on another piece of work)

Feature now Shows on list
image

And it is applied correctly
image

These might be affect (need further testing:

Protector Aasimar: Radiant Soul Damage
MotM Aasimar: Radiant Soul Damage
Cleric: Blessed Strikes
Ranger: Favored Foe
Ranger: Gathered Swarm
Warlock: The Hexblade: Hexblade's Curse
Warlock: The Undead: Grave Touched
Artificer: Battlemaster: Arcane Jolt
Barbarian: Path of the Zealot: Divine Fury
Bard: College of Whispers: Psychic blades
Bloodhunter: Crimson Rite
Cleric: Divine Strike
Fighter: Giant’s Might;
Paladin: Sacred Weapon
Ranger: Gloom Stalker: Dread Ambusher
Ranger: Hunter: Colossus Slayer
Ranger: Monster Slayer: Slayer's Prey
Ranger: Horizon Walker: Planar Warrior
Ranger: Fey Wanderer: Dreadful Strikes
Rogue: Sneak Attack
Warlock: Eldritch Invocation: Lifedrinker

### CODE REVIEW

**Comment by dmportella** [comment]
@kakaroto I will check the other features as time goes on and apply any patches to it, this was working until yesterday but I think they changed it recently

**Review by kakaroto — CHANGES_REQUESTED** [review]
Thanks for reporting and fixing this.
I've just been reviewing it and checking for any other changes needed or affected by this. 
The `featureDetailsToList` method is used though to capture the feature details of all character features: class features, racial traits and feats. It's also used to capture the actions list from the actions tab.
Your change only makes it find the class features, so it will break for the other stuff. I'd suggest the following change to the selector:
```js
const features = $(selector).find(".ct-feature-snippet > .ct-feature-snippet__heading, .ct-feature-snippet--class > div[class*='styles_heading'], .ct-feature-snippet--racial-trait > div[class*='styles_heading'], .ct-feature-snippet--feat > div[class*='styles_heading']")
```
I kept the original selector too because the ct-actions-list still uses the old classes...

However, I also noticed that the `updateFeatures` method below it will now fail to find racial traits because the div has no classes now instead of the `ct-race-detail` it had before 😭 
I've found that this can get us the div we want (line 259): 
```js
const race_detail = $(".ct-features .ct-content-group:has(.ct-feature-snippet--racial-trait)");
```
If you could review my proposed changed and test/confirm it works as expected, then let me know to approve the updated PR.

Thanks!
(by the way, are you in the Beyond20 discord?)

File: src/dndbeyond/base/character.js [hunk 1]
```diff
@@ -222,7 +222,7 @@ class Character extends CharacterBase {
     }
 
     featureDetailsToList(selector, name) {
-        const features = $(selector).find(".ct-feature-snippet > .ct-feature-snippet__heading");
+        const features = $(selector).find(".ct-feature-snippet > .ct-feature-snippet__heading, .ct-feature-snippet--class > div[class*='styles_heading'], .ct-feature-snippet--racial-trait > div[class*='styles_heading'], .ct-feature-snippet--feat > div[class*='styles_heading']")
```
Comment (dmportella) [inline src/dndbeyond/base/character.js:225]: Yep tested it and it all seem to work

File: src/dndbeyond/base/character.js [hunk 2]
```diff
@@ -256,7 +256,7 @@ class Character extends CharacterBase {
             this._class_features = this.getSetting("class-features", []);
         }
 
-        const race_detail = $(".ct-features .ct-race-detail");
+        const race_detail = $(".ct-features .ct-content-group:has(.ct-feature-snippet--racial-trait)");
```
Comment (dmportella) [inline src/dndbeyond/base/character.js:259]: yep this works well too

**Review by dmportella — COMMENTED** [review]
they work as you suggested <3

**Comment by dmportella** [comment]
@kakaroto I tested the changes you suggested and they work well, I had not noticed the other features (it was a bit late at night when i did those changes) but good catch, I was not on the discord, but I have joined it now under the name gothyl

**Comment by dmportella** [comment]
All changes are done as you suggested, i will check if there is any other issues, there is so many "features" on the website that only affect certain classes that takes some time to see what is suppose to happen

**Review by kakaroto — APPROVED** [review]
Thanks for testing and pushing the fix! LGTM!

**Comment by shazam2323** [comment]
I thought this was fixed? As of today it's still not working. My fire genie doesn't have genie's wrath showing

**Comment by kakaroto** [comment]
> I thought this was fixed? As of today it's still not working. My fire genie doesn't have genie's wrath showing

It being fixed doesn't mean it's released and it doesn't mean your extension on your local PC has been updated.
I'll be releasing the update this weekend, then it can take a week or more before the new release is reviewed by the chrome/firefox/edge store teams and it becomes publicly available.
Either be patient, or read here on how to install the development version: https://github.com/kakaroto/Beyond20#developer-mode-installation

**Pull request merged**